test/python/visualization/test_gate_map.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A test for visualizing device coupling maps"""
+import unittest
+import os
+import matplotlib.pyplot as plt
+from qiskit import IBMQ
+from qiskit.test import QiskitTestCase
+from qiskit.visualization.gate_map import _GraphDist, plot_gate_map
+from .visualization import path_to_diagram_reference, QiskitVisualizationTestCase
+
+
+class TestGateMap(QiskitVisualizationTestCase):
+    """ visual tests for plot_gate_map """
+    backend = []
+
+    def setup(self):
+        """ setup for backend """
+        self.backend = IBMQ.backends(filters=lambda x: not x.configuration().simulator)
+
+    def test_plot_gate_map(self):
+        """ tests plotting of gate map of a device (14 qubit and 5 qubit)"""
+        img_ref = path_to_diagram_reference("5bit_quantum_computer.png")
+        for i in self.backend:
+            n = i.configuration().n_qubits
+            if n > 5:
```

Comment:
Could you change this to something like `if n == 14`? This is currently non extensible as if we were to add a device with more qubits this would fail.

Reply:
Yup, have made the changes, included tests for 16 and 20 qubit backends as well. Thank you for pointing this out!
Also noticed there was a tiny mistake in the fake backends where faketokyo had n_qubits set to 16 instead of 20, so I changed that to the appropriate value as well.